project2/src/components/SignUp/index.jsx

```diff
@@ -0,0 +1,100 @@
+import "./style.scss";
+import { Button, Col, Container, Form, Row } from "react-bootstrap";
+import FormGroup from "../FormGroup";
+import { useFormik } from "formik";
+import * as Yup from "yup";
+import { email_regex, password_regex } from "../../constants";
+import { useDispatch } from "react-redux";
+import { signup } from "../../store/Slice/userSlice";
+
+function SignUp() {
+  const dispatch = useDispatch();
+
+  const formik = useFormik({
+    initialValues: {
+      email: "",
+      username: "",
+      password: "",
+      confirm_password: "",
+    },
+    validationSchema: Yup.object({
+      email: Yup.string()
+        .matches(email_regex, "Email không hợp lệ.")
+        .required("Thông tin bắt buộc."),
+      username: Yup.string()
+        .min(2, "Tối thiểu 2 kí tự.")
+        .required("Thông tin bắt buộc."),
+      password: Yup.string()
+        .matches(
+          password_regex,
+          "Mật khẩu tối thiểu 6 kí tự, bao gồm số, chữ hoa, chữ thường."
+        )
+        .required("Thông tin bắt buộc."),
+      confirm_password: Yup.string()
+        .oneOf([Yup.ref("password")], "Mật khẩu không trùng khớp.")
+        .required("Thông tin bắt buộc."),
+    }),
+    onSubmit: (values) => {
+      const { username, email, password } = values;
+      localStorage.setItem("user", JSON.stringify({ username, email }));
+      dispatch(signup({ username, email, password }));
+    },
+  });
+
+  return (
+    <div className="signup">
+      <Container className="signup__wrap">
+        <h1 className="signup__title pb-4">Tạo tài khoản Lazada</h1>
+        <Form onSubmit={formik.handleSubmit} className="signup__form">
+          <Row>
+            <Col md={6}>
+              <FormGroup
+                label="Email*"
```

Comment:
Sau khi sửa những thứ như ở trên, thì thay vì Email*
Em thử như này xem
```suggestion
              <FormGroup
                label="Email"
                required="true"
```